pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -486,6 +481,22 @@ func (c *Compactor) runCompact(ctx context.Context, logger log.Logger, job Job,
 	// It may create blocks with series outside of the fp range of the compactor. Cutting blocks will be addressed in a follow-up PR.
 	metasMatchingJob, blocksMatchingJob := matchingBlocks(metas, job)
 
+	// Tokenizer is not thread-safe so we need one per goroutine.
+	NGramLength := c.limits.BloomNGramLength(job.tenantID)
+	NGramSkip := c.limits.BloomNGramSkip(job.tenantID)
+
+	btReg := prometheus.WrapRegistererWithPrefix("loki_bloom_tokenizer", c.reg)
+	pReg := prometheus.WrapRegistererWith(
+		prometheus.Labels{
+			"component": fmt.Sprintf(
+				"bloom-tokenizer-%s-%s",
+				job.tenantID,
+				job.tableName,
+			),
+		}, btReg)
+
+	bt, _ := v1.NewBloomTokenizer(pReg, NGramLength, NGramSkip)
```

Comment:
I would pass the more important arguments first.

```suggestion
	bt, _ := v1.NewBloomTokenizer(NGramLength, NGramSkip, reg)
```